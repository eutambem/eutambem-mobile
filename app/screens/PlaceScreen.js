import React from 'react';
import PropTypes from 'prop-types';

import { PlacePicker, Container } from '../components/common';

const PlaceScreen = props => (
  <Container innerContainerStyle={{ padding: '2%' }}>
    <PlacePicker
      types={props.navigation.state.params.types}
      onPlaceChange={props.navigation.state.params.onPlaceChange}
    />
  </Container>
);

PlaceScreen.navigationOptions = ({ navigation }) => ({
  headerTitle: navigation.state.params.title,
});

PlaceScreen.propTypes = {
  navigation: PropTypes.shape({
    navigate: PropTypes.func,
    state: PropTypes.shape({
      params: PropTypes.shape({
        types: PropTypes.arrayOf(PropTypes.string).isRequired,
        onPlaceChange: PropTypes.func.isRequired,
      }),
    }),
  }).isRequired,
};

export default PlaceScreen;
