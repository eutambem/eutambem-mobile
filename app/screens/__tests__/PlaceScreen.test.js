import 'react-native';
import React from 'react';
import { shallow } from 'enzyme';
import sinon from 'sinon';

import PlaceScreen from '../PlaceScreen';

const props = {
  navigation: {
    navigate: sinon.spy(),
    state: {
      params: {
        types: ['establishment'],
        onPlaceChange: sinon.spy(),
      },
    },
  },
};

describe('PlaceScreen', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<PlaceScreen {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
