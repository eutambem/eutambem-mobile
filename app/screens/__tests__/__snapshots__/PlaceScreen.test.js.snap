// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlaceScreen renders correctly 1`] = `
<Container
  innerContainerStyle={
    Object {
      "padding": "2%",
    }
  }
>
  <PlacePicker
    onPlaceChange={[Function]}
    types={
      Array [
        "establishment",
      ]
    }
  />
</Container>
`;
